modules/assim.sequential/R/sda.enkf_MultiSite.R

```diff
@@ -738,7 +655,12 @@ sda.enkf.multisite <- function(settings,
       if(adjustment == TRUE){
         analysis <-adj.ens(Pf, X, mu.f, mu.a, Pa)
       } else {
-        analysis <- as.data.frame(mvtnorm::rmvnorm(as.numeric(nrow(X)), mu.a, Pa, method = "svd"))
+        # if we don't have the analysis from the analysis function.
+        if (is.null(enkf.params[[obs.t]]$analysis)) {
+          analysis <- as.data.frame(mvtnorm::rmvnorm(as.numeric(nrow(X)), mu.a, Pa, method = "svd"))
+        } else {
+          analysis <- enkf.params[[obs.t]]$analysis
+        }
```

Comment:
The new analysis function provides samples of analysis ensembles, which are calculated across blocks, and it is faster than sampling everything together, so we don't have to redo the job.